src/writer/blocks/shared_blueprint_registry.py

```
@@ -0,0 +1,388 @@
+            raise ValueError(f"Blueprint {blueprint_dir} has empty block.json")
+        metadata = SharedBlueprintMetadata(**metadata_dict)
+    except json.JSONDecodeError as e:
+        raise ValueError(f"Invalid JSON in {json_path}: {e}") from e
+    except TypeError as e:
+        raise ValueError(f"Invalid metadata structure in {json_path}: {e}") from e
+
+    try:
+        with open(blueprint_path, "r", encoding="utf-8") as f:
+            blueprint_components = json.load(f)
+        if not isinstance(blueprint_components, list):
+            raise ValueError(f"Blueprint {blueprint_dir} blueprint.json must contain a list of components")
```

Ruff (0.14.5)

Prefer `TypeError` exception for invalid type

(TRY004)